### DISCUSSION TITLE
About type `ValueType` in RSuite v5

### DISCUSSION THREAD

**Discussion by vingeraycn** [rsuite/rsuite#1590] [Q&A] [answered]
在一些组件中使用了 `ValueType` 来作为 `value` `defaultValue` 等值相关的类型，但是这个 `ValueType` 一直被作为值类型传入，使用者没有入口去替换值类型。

下面是一张 `SelectPickerProps` 的截图，泛型 `T` 会一直使用缺省值，而使用者没有入口去传入值类型。

In some components, `ValueType` is used as a value-related type such as `value` and `defaultValue`, but this `ValueType` has always been passed in as a value type, and the user has no entry to replace the value type.

Below is a screenshot of `SelectPickerProps`, the generic `T` will always use the default value, and the user has no entry to pass in the value type.

<img width="647" alt="图片" src="https://user-images.githubusercontent.com/19839331/112783360-bfeb9380-9081-11eb-93a5-51a817c540e8.png">

有什么办法可以解决这个问题的吗？

Is there any way to solve this problem?

**Comment by simonguo** [comment, marked as answer]
没了避免复杂的对象类型比较，在 v5 中的 value 限制为值类型 `number | string | (number | string)[]` ...  
用户不能通过 ValueType 去改变 value 的类型。
